test/extensions/filters/http/composite/composite_filter_integration_test.cc

```diff
@@ -139,11 +139,113 @@ class CompositeFilterIntegrationTest : public testing::TestWithParam<Network::Ad
                                                  name));
   }
 
+  void prependCompositeDynamicFilter(const std::string& name = "composite",
+                                     const std::string& path = "set_response_code.yaml") {
+    config_helper_.prependFilter(TestEnvironment::substitute(absl::StrFormat(R"EOF(
+      name: %s
+      typed_config:
+        "@type": type.googleapis.com/envoy.extensions.common.matching.v3.ExtensionWithMatcher
+        extension_config:
+          name: composite
+          typed_config:
+            "@type": type.googleapis.com/envoy.extensions.filters.http.composite.v3.Composite
+        xds_matcher:
+          matcher_tree:
+            input:
+              name: request-headers
+              typed_config:
+                "@type": type.googleapis.com/envoy.type.matcher.v3.HttpRequestHeaderMatchInput
+                header_name: match-header
+            exact_match_map:
+              map:
+                match:
+                  action:
+                    name: composite-action
+                    typed_config:
+                        "@type": type.googleapis.com/envoy.extensions.filters.http.composite.v3.ExecuteFilterAction
+                        dynamic_config:
+                          name: set-response-code
+                          config_discovery:
+                            config_source:
+                                resource_api_version: V3
+                                path_config_source:
+                                  path: "{{ test_tmpdir }}/%s"
+                            type_urls:
+                            - type.googleapis.com/test.integration.filters.SetResponseCodeFilterConfig
+    )EOF",
+                                                                             name, path)));
+    TestEnvironment::writeStringToFileForTest("set_response_code.yaml", R"EOF(
+resources:
+  - "@type": type.googleapis.com/envoy.config.core.v3.TypedExtensionConfig
+    name: set-response-code
+    typed_config:
```

Comment:
YAML has formatting issues?

Reply:
I think it is github view

Reply:
The line numbers are correct though? Maybe YAML parser is lenient and lets you use invalid YAML?

Reply:
May be. but test validates this - so I assume it is correctly being sent to Envoy

Reply:
The test probably uses lenient parser mode. Would be good to clean up for avoid future confusion.

Reply:
oh. did not realize duplication. Was thinking about some thing else. Would clean up